cmd/aro/deploy.go

```diff
@@ -70,28 +70,46 @@ func deploy(ctx context.Context, log *logrus.Entry) error {
 
 	if deployVersion == "unknown" ||
 		(!env.IsLocalDevelopmentMode() && strings.Contains(deployVersion, "dirty")) {
-		return fmt.Errorf("invalid deploy version %q", deployVersion)
+		return nil, fmt.Errorf("invalid deploy version %q", deployVersion)
 	}
 
 	if strings.ToLower(location) != location {
-		return fmt.Errorf("location %s must be lower case", location)
+		return nil, fmt.Errorf("location %s must be lower case", location)
 	}
 
 	config, err := pkgdeploy.GetConfig(flag.Arg(1), location)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	deployer, err := pkgdeploy.New(ctx, log, env, config, deployVersion, tokenCredential)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	err = deployer.PreDeploy(ctx, 30)
+	return deployer, nil
+}
+
+func preDeploy(ctx context.Context, log *logrus.Entry, hasAKS bool) error {
+	deployer, err := getDeployer(ctx, log)
+	if err != nil {
+		return err
+	}
+	err = deployer.PreDeploy(ctx, 30, hasAKS)
 	if err != nil {
 		return err
 	}
 
+	// Must be last step so we can be sure there are no RPs at older versions
+	// still serving
+	return deployer.SaveVersion(ctx)
```

Comment:
After the split of the original function, `SaveVersion` has ended in two places, here and in the new `deploy` function, but this should not be called before we have deployed the RP and Gateway.  This will cause problems in Production, because the new version can be stored even if the actual deploy fails while we still keep the older deploy.

Also, this split is being reflected only in the Makefile, but this refactoring will have implications also in the Production pipelines because we will be only calling `aro deploy`   there. Hence, this invalidates the deployment test we did in INT.